src/main/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestProcessor.java

```diff
@@ -22,44 +28,110 @@
+      CaseRepository caseRepository, EventLogger eventLogger, CaseProcessor caseProcessor) {
     this.caseRepository = caseRepository;
     this.eventLogger = eventLogger;
+    this.caseProcessor = caseProcessor;
   }
 
   public void processFulfilmentRequest(ResponseManagementEvent fulfilmentRequest) {
     EventDTO fulfilmentRequestEvent = fulfilmentRequest.getEvent();
     FulfilmentRequestDTO fulfilmentRequestPayload =
         fulfilmentRequest.getPayload().getFulfilmentRequest();
 
-    String caseId = fulfilmentRequestPayload.getCaseId();
+    Case caze = getCaseByCaseId(fulfilmentRequestPayload);
 
-    Optional<Case> cazeResult = caseRepository.findByCaseId(UUID.fromString(caseId));
+    validateFulfilmentRequiredFields(
+        fulfilmentRequestEvent, fulfilmentRequestPayload, caze.getCaseId().toString());
 
-    if (cazeResult.isEmpty()) {
-      log.error(CASE_NOT_FOUND_ERROR);
-      throw new RuntimeException(String.format("Case ID '%s' not found!", caseId));
+    eventLogger.logFulfilmentRequestedEvent(
+        caze,
+        caze.getCaseId(),
+        fulfilmentRequestEvent.getDateTime(),
+        FULFILMENT_REQUEST_RECEIVED,
+        FULFILMENT_REQUESTED,
+        fulfilmentRequestPayload,
+        fulfilmentRequestEvent);
+
+    if (individualResponseRequestCodes.contains(fulfilmentRequestPayload.getFulfilmentCode())) {
+      Case individualResponseCase = saveIndividualResponseCase(caze);
+      caseProcessor.emitCaseCreatedEvent(individualResponseCase);
     }
+  }
+
+  private Case saveIndividualResponseCase(Case caze) {
+    Case individualResponseCase =
+        caze.toBuilder()
+            .caseId(UUID.randomUUID())
+            .caseRef(caseProcessor.getUniqueCaseRef())
+            .uacQidLinks(null)
+            .events(null)
+            .createdDateTime(OffsetDateTime.now())
+            .state(CaseState.ACTIONABLE)
+            .receiptReceived(false)
+            .refusalReceived(false)
+            .addressType(HOUSEHOLD_INDIVIDUAL_RESPONSE_ADDRESS_TYPE)
+            .addressLevel(null)
+            .htcWillingness(null)
+            .htcDigital(null)
+            .fieldCoordinatorId(null)
+            .fieldOfficerId(null)
+            .treatmentCode(null)
+            .ceExpectedCapacity(null)
+            .build();
+
+    caseRepository.save(individualResponseCase);
+
+    return individualResponseCase;
+  }
 
-    if (fulfilmentRequestEvent.getDateTime() == null) {
+  private void validateFulfilmentRequiredFields(
+      EventDTO eventDTO, FulfilmentRequestDTO fulfilmentRequestDTO, String caseId) {
+    if (eventDTO.getDateTime() == null) {
       log.error(DATETIME_NOT_PRESENT);
       throw new RuntimeException(
           String.format("Date time not found in fulfilment request event for Case ID '%s", caseId));
     }
 
-    Case caze = cazeResult.get();
+    if (StringUtils.isEmpty(fulfilmentRequestDTO.getFulfilmentCode())) {
+      log.error(FULFILMENT_CODE_NOT_FOUND);
```

Comment:
it's not "not found" it's "not present" i.e. null or blank, which is different. This kind of validation is nonsense... if you send us a garbage message, tough luck buddy. Otherwise we'll be spending an infinite amount of time imagining all the myriad ways that upstream could possibly mess up, which is clearly out of scope for a closed system

Reply:
removed